tests/test_codex_feed_hooks.py

```
@@ -579,6 +579,317 @@ def test_codex_monitor_survives_transient_owner_rpc_timeout(cli_path: str, root:
+            run_hook(["hooks", "codex", "stop"], base_payload)
+            time.sleep(0.2)
+            premature_commands = raw_commands_after(fake, before_stop)
+            if any(
+                "set_agent_lifecycle codex idle" in command
+                or command.startswith("notify_target_async ")
+                for command in premature_commands
+            ):
+                raise AssertionError(
+                    "Codex Stop finalized while structured background work "
+                    f"was active: {premature_commands!r}"
+                )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)